tests/e2e/runner/__tests__/pmp/pmp.test.ts

```diff
@@ -0,0 +1,434 @@
+
+    // re-configure with different parameter
+    const newParamKey = "size";
+    tx = await pmpV0Contract
+      .connect(artist)
+      .configureProject(genArt721CoreAddress, projectId, [
+        {
+          key: newParamKey,
+          pmpConfig: {
+            authOption: 3, // ArtistAndTokenOwner auth
+            paramType: 3, // Uint256Range type
+            pmpLockedAfterTimestamp: 0,
+            authAddress: ethers.constants.AddressZero,
+            selectOptions: [],
+            minRange:
+              "0x0000000000000000000000000000000000000000000000000000000000000001", // 1
+            maxRange:
+              "0x000000000000000000000000000000000000000000000000000000000000000a", // 10
+          },
+        },
+      ]);
+    const receiptTwo = await tx.wait();
+    const projectConfiguredNextTimestamp = (
+      await artist.provider.getBlock(receiptTwo.blockNumber)
+    )?.timestamp;
+
+    await waitUntilSubgraphIsSynced(client);
+
+    const projectPmpConfigNextRes = await getProjectPMPConfigDetails(
+      client,
+      targetId
+    );
+    const projectNextRes = await getProjectDetails(client, fullProjectId);
+
+    // verify only one config exists and it's the new one
+    const projectPmpConfigsNext = projectPmpConfigNextRes?.pmpConfigs;
+    expect(projectPmpConfigsNext?.length).toBe(1);
+    expect(projectPmpConfigNextRes?.pmpConfigCount).toBe("1");
+
+    // verify the new config details
+    const nextConfig = projectPmpConfigsNext?.[0];
+    expect(nextConfig?.pmpProjectConfig?.id).toBe(targetId);
+    expect(nextConfig?.key).toBe(newParamKey);
+    expect(nextConfig?.authOption).toBe("ArtistAndTokenOwner");
+    expect(nextConfig?.paramType).toBe("Uint256Range");
+    expect(nextConfig?.selectOptions).toEqual([]);
+    expect(nextConfig?.minRange).toBe(
+      "0x0000000000000000000000000000000000000000000000000000000000000001"
+    );
+    expect(nextConfig?.maxRange).toBe(
+      "0x000000000000000000000000000000000000000000000000000000000000000a"
+    );
+    expect(nextConfig?.createdAt).toBe(
+      projectConfiguredNextTimestamp.toString()
+    );
+    expect(projectPmpConfigNextRes?.updatedAt).toBe(
+      projectConfiguredNextTimestamp.toString()
+    );
+
+    // verify the old config key is no longer in the keys array
+    const oldPmpConfigKey = `${pmpV0Address.toLowerCase()}-${genArt721CoreAddress.toLowerCase()}-${projectId}-${firstParamKey}`;
+    const newPmpConfigKey = `${pmpV0Address.toLowerCase()}-${genArt721CoreAddress.toLowerCase()}-${projectId}-${newParamKey}`;
+    expect(projectPmpConfigNextRes?.pmpConfigKeys).toContain(newPmpConfigKey);
+    expect(projectPmpConfigNextRes?.pmpConfigKeys).not.toContain(
+      oldPmpConfigKey
+    );
+
+    // validate project field
+    expect(projectNextRes?.pmpProjectConfig?.id).toBe(targetId);
+  });
+
+  describe("TokenParamsConfigured", () => {
+    test("creates a new PMP config for a token when configured", async () => {
```

Comment:
it might be worth checking another case for when a PMP config is created for a non-existent token?

Reply:
Added!